pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/xpath/saxon/ElementNode.java

```diff
@@ -147,10 +157,18 @@ public Value atomize() {
     }
 
     @Override
-    public CharSequence getStringValueCS() {
+    public String getStringValue() {
+        if (determineType(getUnderlyingNode()) == Type.TEXT) {
```

Comment:
Comment nodes should also take this path: https://www.w3.org/TR/1999/REC-xpath-19991116/#section-Comment-Nodes

Also the kind has already been computed
```suggestion
        if (getNodeKind() == Type.TEXT || getNodeKind() == Type.COMMENT) {
```